### ISSUE TITLE
Compile error on ghc-8.2.2

### ISSUE THREAD

**Issue by xaviershay** [goldfirere/units#57]
Using stack to install I get the following error:

```
--  While building custom Setup.hs for package units-2.4 using:
      /Users/xavier/.stack/setup-exe-cache/x86_64-osx/Cabal-simple_mPHDZzAJ_2.0.1.0_ghc-8.2.2 --builddir=.stack-work/dist/x86_64-osx/Cabal-2.0.1.0 build --ghc-options " -ddump-hi -ddump-to-file -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/xavier/Personal/haskell-sandbox/.stack-work/logs/units-2.4.log

    Configuring units-2.4...
    Preprocessing library for units-2.4..
    Building library for units-2.4..
    [ 1 of 19] Compiling Data.Metrology.Set ( Data/Metrology/Set.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Set.o )
    [ 2 of 19] Compiling Data.Metrology.Z ( Data/Metrology/Z.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Z.o )
    [ 3 of 19] Compiling Data.Metrology.Factor ( Data/Metrology/Factor.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Factor.o )
    [ 4 of 19] Compiling Data.Metrology.LCSU ( Data/Metrology/LCSU.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/LCSU.o )
    [ 5 of 19] Compiling Data.Metrology.Dimensions ( Data/Metrology/Dimensions.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Dimensions.o )
    [ 6 of 19] Compiling Data.Metrology.Units ( Data/Metrology/Units.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Units.o )
    [ 7 of 19] Compiling Data.Metrology.Qu ( Data/Metrology/Qu.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Qu.o )
    [ 8 of 19] Compiling Data.Metrology.Unsafe ( Data/Metrology/Unsafe.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Unsafe.o )
    [ 9 of 19] Compiling Data.Metrology.Show ( Data/Metrology/Show.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Show.o )
    [10 of 19] Compiling Data.Metrology.Combinators ( Data/Metrology/Combinators.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Combinators.o )
    [11 of 19] Compiling Data.Metrology.Validity ( Data/Metrology/Validity.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Validity.o )
    [12 of 19] Compiling Data.Metrology.Linear ( Data/Metrology/Linear.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Linear.o )
    [13 of 19] Compiling Data.Metrology.Internal ( Data/Metrology/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Internal.o )
    [14 of 19] Compiling Data.Metrology.Vector ( Data/Metrology/Vector.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Vector.o )
    [15 of 19] Compiling Data.Metrology.Poly ( Data/Metrology/Poly.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/Poly.o )
    [16 of 19] Compiling Data.Metrology.TH ( Data/Metrology/TH.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Metrology/TH.o )
    
    /private/var/folders/mx/91rymfz14gl9r0ffxhf2j9d8000152/T/stack36007/units-2.4/Data/Metrology/TH.hs:222:69: error:
        • Couldn't match type ‘Type’ with ‘DerivClause’
          Expected type: [DerivClause]
            Actual type: [Type]
        • In the sixth argument of ‘DataD’, namely ‘(map ConT derivs)’
          In the expression:
            DataD ct name tvbs Nothing cons (map ConT derivs)
          In an equation for ‘mkDataD’:
              mkDataD ct name tvbs cons derivs
                = DataD ct name tvbs Nothing cons (map ConT derivs)
        |
    222 | mkDataD ct name tvbs cons derivs = DataD ct name tvbs Nothing cons (map ConT derivs)
        |    
```

Sorry I'm too new to haskell to be much help in fixing :(

**Comment by goldfirere** [comment]
I've just uploaded a new version. I believe that should fix the problem. (I'm afraid I don't know `stack` well enough to give specific instructions for how to access this new upload. You'll have to get `stack` to refresh its package list from hackage.haskell.org.)

I had actually fixed this problem months ago, but never fixed the last niggling validation error and uploaded. It's all set now, in any case.

Thanks for posting!
